pkg/admin/service/interaction.go

```diff
@@ -0,0 +1,43 @@
+package service
+
+import "github.com/authgear/authgear-server/pkg/lib/interaction"
+
+type InteractionGraphService interface {
+	NewGraph(ctx *interaction.Context, intent interaction.Intent) (*interaction.Graph, error)
+	DryRun(webStateID string, fn func(*interaction.Context) (*interaction.Graph, error)) error
+	Run(webStateID string, graph *interaction.Graph, preserveGraph bool) error
+}
+
+type InteractionService struct {
+	Graph InteractionGraphService
+}
+
+func (s *InteractionService) Perform(intent interaction.Intent, input interface{}) (*interaction.Graph, error) {
```

Comment:
Do we need to adjust the semantics of this function so that after this function returns, the graph has finished? Right now the sole caller ignore the graph with `_`, which may lead to unexpected result in the future.

Reply:
That's the intention. If the graph has not finished, `DryRun`/`Run` would return an error, so as long as `Perform` returns without errors, the graph is finished.

The graph is returned in case API may need to extract information from it (e.g. created user ID).